mm2src/coins/lightning.rs

```diff
@@ -643,38 +773,190 @@ pub async fn open_channel(ctx: MmArc, req: OpenChannelRequest) -> OpenChannelRes
         user_config.own_channel_config.our_htlc_minimum_msat = min;
     }
 
+    let rpc_channel_id = ln_coin.persister.get_last_channel_rpc_id().await? as u64 + 1;
+
     let temp_channel_id = async_blocking(move || {
         channel_manager
-            .create_channel(node_pubkey, amount_in_sat, push_msat, 1, Some(user_config))
+            .create_channel(node_pubkey, amount_in_sat, push_msat, rpc_channel_id, Some(user_config))
             .map_to_mm(|e| OpenChannelError::FailureToOpenChannel(node_pubkey.to_string(), format!("{:?}", e)))
     })
     .await?;
 
     {
-        let mut unsigned_funding_txs = ln_coin.platform_fields.unsigned_funding_txs.lock();
-        unsigned_funding_txs.insert(temp_channel_id, unsigned);
+        let mut unsigned_funding_txs = ln_coin.platform.unsigned_funding_txs.lock();
+        unsigned_funding_txs.insert(rpc_channel_id, unsigned);
     }
 
+    let pending_channel_details = SqlChannelDetails::new(
+        rpc_channel_id,
+        temp_channel_id,
+        node_pubkey,
+        true,
+        user_config.channel_options.announced_channel,
+    );
+
     // Saving node data to reconnect to it on restart
-    ln_coin.nodes_addresses.lock().insert(node_pubkey, node_addr);
-    ln_coin.persister.save_nodes_addresses(ln_coin.nodes_addresses).await?;
+    ln_coin.open_channels_nodes.lock().insert(node_pubkey, node_addr);
+    ln_coin
+        .persister
+        .save_nodes_addresses(ln_coin.open_channels_nodes)
+        .await?;
+
+    ln_coin.persister.add_channel_to_sql(pending_channel_details).await?;
 
     Ok(OpenChannelResponse {
-        temporary_channel_id: temp_channel_id.into(),
+        rpc_channel_id,
         node_address: req.node_address,
     })
 }
 
 #[derive(Deserialize)]
-pub struct ListChannelsRequest {
+pub struct OpenChannelsFilter {
+    pub channel_id: Option<H256Json>,
+    pub counterparty_node_id: Option<PublicKeyForRPC>,
+    pub funding_tx: Option<H256Json>,
+    pub from_funding_value_sats: Option<u64>,
+    pub to_funding_value_sats: Option<u64>,
+    pub is_outbound: Option<bool>,
+    pub from_balance_msat: Option<u64>,
+    pub to_balance_msat: Option<u64>,
+    pub from_outbound_capacity_msat: Option<u64>,
+    pub to_outbound_capacity_msat: Option<u64>,
+    pub from_inbound_capacity_msat: Option<u64>,
+    pub to_inbound_capacity_msat: Option<u64>,
+    pub confirmed: Option<bool>,
+    pub is_usable: Option<bool>,
+    pub is_public: Option<bool>,
+}
+
+fn apply_open_channel_filter(channel_details: &ChannelDetailsForRPC, filter: &OpenChannelsFilter) -> bool {
+    let is_channel_id = if let Some(channel_id) = filter.channel_id {
```

Comment:
Are all these filters really required? Also, 
```rust
let is_channel_id = filter.channel_id.is_none() || Some(&channel_details.channel_id) == filter.channel_id.as_ref();
```
is much easier to understand and read, I think.

Reply:
> Are all these filters really required?

I guess the most important filters are `is_outbound`, `is_confirmed`, `is_usable`, `is_public`. 
`channel_id`, `counterparty_node_id`, `funding_tx` can be important too as filtering here is like a search.

As for the rest, I added them since filtering on any parameter is optional anyway so I thought to have them just in case they are needed. I think for balance, capacity etc... sorting can be more desired similar to [here](https://1ml.com/testnet/node/0374c403a1f2733ae30d466c9e1c1a15e10bab46cb1b17cb4d3ec6db618c92545b/channels?order=newest), but maybe we should wait and see what the GUI will need.